src/ExtraPackage.php

```diff
@@ -644,6 +646,51 @@ public static function unwrapIfNeeded(
         return $root;
     }
 
+    protected function mergeAliases(RootPackageInterface $root)
+    {
+        $aliases = array();
+        $unwrapped = $this->unwrapIfNeeded($root, 'setAliases');
+        foreach (array('require', 'require-dev') as $linkType) {
+            $linkInfo = BasePackage::$supportedLinkTypes[$linkType];
+            $method = 'get'.ucfirst($linkInfo['method']);
+            $links = array();
+            foreach ($unwrapped->$method() as $link) {
+                $links[$link->getTarget()] = $link->getConstraint()->getPrettyString();
+            }
+            $aliases = $this->extractAliases($links, $aliases);
+        }
+        $unwrapped->setAliases($aliases);
+    }
+
+    /**
+     * Extract aliases from version constraints (dev-branch as 1.0.0).
+     *
+     * @param array $requires
+     * @param array $aliases
+     * @return array
+     * @see RootPackageLoader::extractAliases()
+     */
+    protected function extractAliases(array $requires, array $aliases)
```

Comment:
Annoying that we can't just use the `RootPackageLoader::extractAliases()` instead of having to duplicate it